### PR TITLE
DRAFT: attempt to fix hang on voices for espeak in test

### PR DESCRIPTION
**Pull request by willwade**
@cclauss take a look - i think this fixes this hang on windows.. but maybe ive refactored too much

### CODE REVIEW

**Comment by willwade** [comment]
nb that assertion on length is the expected number of voices. Im nervous about that - maybe it should just be 

`assert len(voices) > 10, f"Expected more than 10 voices in {engine}, but found {len(voices)}"`

File: tests/test_engines.py [hunk 1]
```diff
@@ -38,33 +38,33 @@ def test_espeak_voices(driver_name):
...
-            voice == "English (Great Britain)"
-        ), f"Expected {engine} default voice to be 'English (Great Britain)'"
+    assert str(engine) == "espeak", "Expected engine name to be 'espeak'"
+
+    # Retrieve and print voices without modifying `voice` property
     voices = engine.getProperty("voices")
-    print(f"{engine} has {len(voices) = } voices.")
-    # Linux eSpeak-NG v1.50 has 109 voices,
-    # macOS eSpeak-NG v1.51 has 131 voices,
-    # Windows eSpeak-NG v1.52-dev has 221 voices.
```
Comment (cclauss) [inline tests/test_engines.py:53]: The comments are helpful.

Comment (cclauss) [reply tests/test_engines.py:53]: Tests should be verbose...  Lots of checks.  We should not optimize them.

File: tests/test_engines.py [hunk 2]
```diff
@@ -38,33 +38,33 @@ def test_espeak_voices(driver_name):
...
+        109,
+        131,
+        221,
+    }, f"Expected 109, 131, or 221 voices in {engine}"
+
+    # Filter English voices
     english_voices = [voice for voice in voices if voice.id.startswith("English")]
-    # Linux eSpeak-NG v1.50 has 7 English voices,
-    # macOS eSpeak-NG v1.51 and Windows eSpeak-NG v1.52-dev have 8 English voices.
-    assert len(english_voices) in {7, 8}, "Expected 7 or 8 English voices in {engine}"
```
Comment (cclauss) [inline tests/test_engines.py:59]: The comments are useful documentation.

File: tests/test_engines.py [hunk 3]
```diff
@@ -38,33 +38,33 @@ def test_espeak_voices(driver_name):
...
+    assert len(english_voices) in {7, 8}, "Expected 7 or 8 English voices"
+
+    # Queue a single utterance for each voice without calling `runAndWait()` in a loop
     names = []
     for _voice in english_voices:
         engine.setProperty("voice", _voice.id)
-        # English (America, New York City) --> America, New York City
```
Comment (cclauss) [inline tests/test_engines.py:63]: This is much better documentation of what the code does than ~ Extract hame for display.

File: tests/test_engines.py [hunk 4]
```diff
@@ -73,8 +73,8 @@ def test_espeak_voices(driver_name):
     no_nyc = expected.rpartition("|")[0]
     assert name_str in {expected, no_nyc}, f"Expected '{expected}' or '{no_nyc}'."
     print(f"({name_str.replace('|', ' ; ')})", end=" ", flush=True)
-    engine.runAndWait()
-    engine.setProperty("voice", voice)  # Reset voice to original value
```
Comment (cclauss) [inline tests/test_engines.py:77]: Listen to the tests that come after this one to see why restoring defaults makes things more intuitive.

**Comment by willwade** [comment]
NB: IT seems we have a problem in espeak set voice..

**Review by cclauss — APPROVED** [review]
Looks great!

**Comment by cclauss** [comment]
?

We need to get the Windows test passing soon.

**Pull request closed without merging**